docs/HowToGuides/GettingStarted.md

```diff
@@ -190,6 +190,9 @@ toolchain as a one-off, there are a couple of differences:
 > LLDB currently requires at least `swig-1.3.40` but will successfully build
 > with version 2 shipped with Ubuntu.
 
+> **Note**  
+> When building on Linux, make sure to have at least 8 GB of RAM when using flag `--release-debuginfo` and 32 GB of RAM when using flag `--debug-swift`, plus 150 GB of disk space available.
+
```

Comment:
* I think this should go under "System Requirements". Perhaps we should update the default disk space requirement to 100GB too.
* What’s taking up 150GB on Linux? The entire checkout plus a `--release-debuginfo --debug-swift` build is only about 70GB for me on macOS.
* How much RAM was not enough to pull through a `--debug-swift` build, 16GB?

Reply:
> I think this should go under "System Requirements". Perhaps we should update the default disk space requirement to 100GB too.

Amazing, I did not even notice this section! Surely, updated 🤝 

> What’s taking up 150GB on Linux? The entire checkout plus a --release-debuginfo --debug-swift build is only about 70GB for me on macOS.

--debug-swift took 78 GB of disk space, plus according to [Development Tips](https://github.com/apple/swift/blob/main/docs/DevelopmentTips.md) it is recommended to set sccache `SCCACHE_CACHE_SIZE` environment variable to 50 GB. Then, system size is about 19 GB. 
I've ran out of space mostly because I am still unable to compile with `--debug-swift --debug-foundation` flags, and have to resort to `--release-debuginfo -T --debug-swift-stdlib` to make some progress in crash investigation. Nonetheless, 150 GB is enough to avoid running into a problem with low disk space in the middle of LLVM compilation, for example.

|Folders|Size|
:-:|:-:
<img width="427" alt="image" src="https://github.com/apple/swift/assets/1614869/a57f7195-5cd9-4377-b462-040ec6d00e61">|<img width="253" alt="image" src="https://github.com/apple/swift/assets/1614869/d1485575-78e1-42c5-a26d-fea1cf274a9d">

> How much RAM was not enough to pull through a --debug-swift build, 16GB?
Answering your question, 4 GB was my initial VM setup, and I wasted 3 days figuring out why linking was failing, discovered the fact by accident. So my intention with this PR is to let others know how to avoid redundant investigations and work.

Even with 32 GB of RAM, linking of `clang-17` binary fails every single time in a VM. On hardware Linux, it'd be better I presume; however, after 2-3 tries I am able to link largest binaries. On the other hand, if to limit concurrent tasks with `-j 5`, for instance, failures are more rare, but still occur for the same reason, not enough RAM available.